packages/runtime/src/templates/getPageResolver.ts

```diff
@@ -12,19 +13,27 @@ import { getDependenciesOfFile } from '../helpers/files'
 // build. This is used by the nft bundler to find all the pages.
 
 export const getPageResolver = async ({ publish, target }: { publish: string; target: string }) => {
-  const functionDir = posix.resolve(posix.join('.netlify', 'functions', HANDLER_FUNCTION_NAME))
-  const root = posix.resolve(slash(publish), target === 'server' ? 'server' : 'serverless', 'pages')
-
-  const pages = await glob('**/*.js', {
+  const functionDir = resolve(join('.netlify', 'functions', HANDLER_FUNCTION_NAME))
+  const root = resolve(slash(publish), target === 'server' ? 'server' : 'serverless')
+  const pages = await globby('{pages,app}/**/*.js.nft.json', {
     cwd: root,
     dot: true,
   })
-  const pageFiles = pages
-    .map((page) => `require.resolve('${posix.relative(functionDir, posix.join(root, slash(page)))}')`)
-    .sort()
+
+  const dependencies = await Promise.all(
+    pages.map(async (page) => {
+      const dir = dirname(page)
+      const { files } = await readJSON(join(root, page))
+      return files?.map((file) => resolve(root, dir, file))
+    }),
+  )
+
+  const deduped = [...new Set(dependencies.flat())]
+
+  const pageFiles = deduped.map((file) => `require.resolve('${relative(functionDir, file)}')`).sort()
 
   return outdent`
-    // This file is purely to allow nft to know about these pages. It should be temporary.
+    // This file is purely to allow nft to know about these pages.
```

Comment:
I guess it's no longer temporary? 😎

Reply:
Well, it's been a year and there's no sign of custom nft support being prioritised, so this is more wishful thinking!